### PR TITLE
Remove world-writability from {profiles,gcroots}/per-user 

### PR DESCRIPTION
**Pull request by edolstra**
These directories are now created on demand by the daemon.

Also some simplifications to the profile scripts.

Alternative to #3134, #3135.

### CODE REVIEW

File: scripts/install-multi-user.sh [hunk 1]
```diff
@@ -437,24 +437,17 @@ create_build_users() {
 }
 
 create_directories() {
```
Comment (matthewbauer) [inline scripts/install-multi-user.sh]: We probably need to think about how this effect current installs. Should Nix include some migration for them?

Comment (edolstra) [reply scripts/install-multi-user.sh]: No, `LocalStore::LocalStore()` automatically fixes permissions on `{profiles,gcroots}/per-user`.

File: scripts/nix-profile-daemon.sh.in [hunk 2]
```diff
@@ -21,18 +15,6 @@ if test -w $HOME; then
       fi
   fi
 
-  # Subscribe the root user to the NixOS channel by default.
-  if [ "$USER" = root -a ! -e $HOME/.nix-channels ]; then
```
Comment (matthewbauer) [inline scripts/nix-profile-daemon.sh.in]: This may be worth keeping. I'm not sure the original use case, but it doesn't seem less relevant now.

Comment (edolstra) [reply scripts/nix-profile-daemon.sh.in]: It's already done by the installer script, so we don't need to check for it here.

**Comment by grahamc** [comment]
The install matrix failed on almost all cases: https://buildkite.com/organizations/grahamc/pipelines/nix-install-matrix/builds/33/jobs/b20b2d91-a52a-41be-b210-ca8b6a5709eb/artifacts/bdfccc3f-8bd0-4917-b266-96a5a0ce0b23

Seems the problem is quite consistently:

> error: opening lock file '/nix/var/nix/db/big-lock': Permission denied

I ran this after the first push. I'll re-run (but on a bigger server so it doesn't take 2h...)

**Comment by edolstra** [comment]
Where are you seeing that? Your link shows a lot of "command not found" errors.

**Comment by grahamc** [comment]
Oops, corrected link: https://buildkite.com/organizations/grahamc/pipelines/nix-install-matrix/builds/72/jobs/d3c62ce0-0303-45ef-8120-4fb8efae434e/artifacts/d6e535d9-2b17-470d-8722-d63dbece492e

**Comment by grahamc** [comment]
Looks good from the install test matrix:

2.3: https://buildkite.com/organizations/grahamc/pipelines/nix-install-matrix/builds/33/jobs/b20b2d91-a52a-41be-b210-ca8b6a5709eb/artifacts/bdfccc3f-8bd0-4917-b266-96a5a0ce0b23
this new build: https://buildkite.com/organizations/grahamc/pipelines/nix-install-matrix/builds/73/jobs/1c0b0150-92d2-4a5f-a024-b986f7b76432/artifacts/527f7457-2a7b-45ed-b220-681068399d04

**Comment by grahamc** [comment]
_pro tip: open them in two windows next to each other, and zoom out of the page as much as possible to compare._

**Comment by alyssais** [comment]
Can we please have the CVE in at least one of the commit messages, for future reference?

**Pull request merged**